pcdsdevices/epics/mps.py

```diff
@@ -101,3 +101,36 @@ def _fault_change(self, *args, **kwargs):
         """
         kwargs.pop('sub_type', None)
         self._run_subs(sub_type=self.SUB_FAULT_CH, **kwargs)
+
+
+def mps_factory(clsname, cls,  *args, mps_prefix=None, veto=False,  **kwargs):
+    """
+    Create a new object of arbitrary class capable of storing MPS information
+
+    A new class identical to the provided one is created, but with additional
+    attribute `mps` that relies upon the provided `mps_prefix`. All other
+    information is passed through to the class constructor as args and kwargs
+
+    Parameters
+    ----------
+    clsname : str
+        Name of new class to create
+
+    cls :
+        Device class to add `mps`
+
+    mps_prefix : str
+        Prefix for MPS subcomponent
+
+    veto : bool, optional
+        Whether the MPS bit is capable of veto
+
+    args :
+        Passed to device constructor
+
+    kwargs:
+        Passed to device constructor
+    """
+    comp = FC(MPS, mps_prefix, veto=veto)
```

Comment:
Does this need to be a FC?

Reply:
Thought this was the way to not add the master `prefix`?